designs/inter-ident-verify.md

```diff
@@ -0,0 +1,78 @@
+9. A decrypts result with A private key from server X.
+10. A decrypts result with U public key from message. A now has the original message.
+
+11. A encrypts result with A private key from server X, acting as a signature from A directly.
+12. A encrypts result with user public key from message, preventing S from decrypting the original message.
+13. A encrypts result with S public key from server X, verifying validity of S.
+14. A sends result to S.
+
+#### Trusted caller (S)
+15. S decrypts message with S private key from server Y, and can't decrypted further.
+16. S sends result to U via phone call.
+
+#### Individual user (U)
+17. U decrypts result with U private key from server Y.
+18. U decrypts message with A public key from server Y.
+19. U verifies that result is the same as the original message generated.
+
+It's important to note that S has no obligation to send U's public key to A, but then A would not be able to recover the original message, and the verification would fail because garbage in, garbage out. Therefore, the verification can only pass if S sends a valid copy of U's public key to A. Following this algorithm, S can never decrypt the message being sent between the U and the A, and the A verifies and signs U's unique message. 
+
+The key aspect here is that the message being sent and received by U is the same, but encrypted in a different order. This means that it's essential to use a non-communitive cryptosystem.
+___
+## Readable hashes
+An important aspect of this process is that users are able to read their hashes over the phone. To increase the readability of these hashes, I propose a modified number representation system similar to hexadecimal where each digit has 64 variants, `{a-z, A-Z, 0-9, *, #}`, allowing 6-bits of information to be stored per digit. With only 6 digits in this number system, we can represent up to 68,719,476,736, covering all 32-bit numbers. With 11 digits, we can represent up to 7.38e19, covering all 64-bit numbers. For shorter messages, this should be enough to read these values over voice.
+___
+## User interface
+When the user receives a call, they ask for the callers root company. They then navigate and log into the Kanidm web interface on their smartphone, running on server Y, search up the root company, and click a button that does steps 1-3, and displays the result to the user in the number system specified above. This hash is displayed to the user, where they can transmit the code directly over the phone since it should be relatively short. The caller enters these numbers into their end, where steps 5-15 are performed remotely from the user. Once the caller reads the resulting hash back to the user, they will enter this back into the web interface, which performs steps 17-19, and tells the user if the verification was successful. This allows the user to verify that the caller on the other end of the phone is the company that the corporation sees and validates.
```

Comment:
Specified how reading the numbers is related to steps. Does this make sense / how can I clarify here?

Reply:
I think you need to do a second, stepped through worked example.